src/KubernetesClient/KubernetesClientConfiguration.ConfigFile.cs

```diff
@@ -557,13 +558,38 @@ public static ExecCredentialResponse ExecuteExternalCommand(ExternalExecution co
                 throw new KubeConfigException($"external exec failed due to: {ex.Message}");
             }
 
-            var stdout = process.StandardOutput.ReadToEnd();
-            var stderr = process.StandardError.ReadToEnd();
-            if (string.IsNullOrWhiteSpace(stderr) == false)
+            var sb = new StringBuilder();
+            var buffer = new char[1024];
+            while (true)
+            {
+                var readTask = process.StandardError.ReadAsync(buffer, 0, buffer.Length);
+
+                if (readTask.Wait(TimeSpan.FromSeconds(2)))
+                {
+                    var bytesRead = readTask.Result;
+                    if (bytesRead == 0)
+                    {
+                        break; // end of stream reached
+                    }
+
+                    sb.Append(buffer, 0, bytesRead);
+                }
+                else
+                {
+                    // timeout occurred
+                    break;
+                }
+            }
+
+            var stderr = sb.ToString();
+
+            if (!string.IsNullOrWhiteSpace(stderr))
             {
                 throw new KubeConfigException($"external exec failed due to: {stderr}");
             }
 
+            var stdout = process.StandardOutput.ReadToEnd();
+
             // Wait for a maximum of 5 seconds, if a response takes longer probably something went wrong...
             process.WaitForExit(5);
```

Comment:
maybe it is to move wait output to a task and put a overall timeout here

Reply:
I added a 5s timeout to the StandardOutput.